packages/create-react/src/components/views/RiversView.tsx

```diff
@@ -0,0 +1,298 @@
+import {
+  Color,
+  MapView,
+  MapViewState,
+  WebMercatorViewport,
+} from '@deck.gl/core';
+import { useContext, useEffect, useMemo, useState } from 'react';
+import { AppContext } from '../../context';
+import { useDebouncedState } from '../../hooks/useDebouncedState';
+import {
+  createViewportSpatialFilter,
+  vectorTilesetSource,
+} from '@carto/api-client';
+import { BASEMAP, VectorTileLayer } from '@deck.gl/carto';
+import { Card } from '../Card';
+import { FormulaWidget } from '../widgets/FormulaWidget';
+import DeckGL from '@deck.gl/react';
+import { Map } from 'react-map-gl/maplibre';
+import { Layers } from '../Layers';
+import { HistogramWidget } from '../widgets/HistogramWidget';
+import { LegendEntryCategorical } from '../legends/LegendEntryCategorical';
+
+const CONNECTION_NAME = 'amanzanares-pm-bq';
+const TILESET_NAME =
+  'cartodb-on-gcp-pm-team.amanzanares_opensource_demo.national_water_model_tileset_final_test_4';
+const MAP_VIEW = new MapView({ repeat: true });
+
+const INITIAL_VIEW_STATE: MapViewState = {
+  latitude: 30.5,
+  longitude: -90.1,
+  zoom: 6,
+};
+
+function hexToRgb(hex: string) {
+  const r = parseInt(hex.slice(1, 3), 16);
+  const g = parseInt(hex.slice(3, 5), 16);
+  const b = parseInt(hex.slice(5, 7), 16);
+  return [r, g, b];
+}
+
+const colors = [
+  '#08589e',
+  '#2b8cbe',
+  '#4eb3d3',
+  '#7bccc4',
+  '#a8ddb5',
+  '#ccebc5',
+  '#e0f3db',
+  '#f7fcf0',
+].map((hex) => hexToRgb(hex));
+
+function streamOrderToColor(n: number, colors: number[][]) {
+  const rgb = colors[Math.min(n - 1, 7)];
+  const alpha = Math.min(50 + n * 20, 255); // Gradually increases opacity with stream order
+  return new Uint8Array([...rgb, alpha]);
+}
+
+/**
+ * MINIMUM STREAM ORDER
+ * Our tileset was generated in a way that it drops water streams of low orders at low zoom levels.
+ * Let's add logic in our app to handle this.
+ */
+function getMinStreamOrder(zoomLevel: number) {
+  if (zoomLevel < 5.5) {
+    return 4; // at zoom 5.5, this tileset only uses streams of order 4 and above
+  } else if (zoomLevel < 6.5) {
+    return 3; // at zoom 6.5, this tileset only uses streams of order 3 and above
+  } else if (zoomLevel < 7.5) {
+    return 2; // at zoom 7.5, this tileset only uses streams of order 2 and above
+  } else {
+    return 1; // at zoom 10, we show all streams
+  }
+}
+
+const MAX_STREAM_ORDER = 10;
+
+/**
+ * Example application page, showing U.S. streams network.
+ */
+export default function IncomeView() {
```

Comment:
Minor: old name here, should be RiversView

Reply:
thanks for the catch here